### PR TITLE
Switch to Pyhon3 plus some groundwork for #88

### PR DESCRIPTION
**Pull request by oschaaf**
Switch to Pyhon3 plus some groundwork for #88

- Switches the integration tests to use python 3 
- Update the CI build image
- Use RAII in python socket handling as that now becomes possible
- Adds some things we'll need in https://github.com/envoyproxy/nighthawk/pull/88

### CODE REVIEW

File: integration/BUILD [hunk 1]
```diff
@@ -12,4 +20,16 @@ py_test(
         "//:nighthawk_test_server",
         "@envoy//test/config/integration/certs",
     ],
+    # We enforce python 2, but I we may be able to stop doing that when we update
+    # bazel in CI. (Or we add python3 explicitly as a build dependency somehow).
```
Comment (htuch) [inline integration/BUILD]: Bazel is now at 0.27 on the latest hash, there is no PY2 here, stale comment?

File: integration/integration_test_fixtures.py [hunk 2]
```diff
@@ -43,11 +45,10 @@ def getFreeListenerPortForAddress(self, address):
...
     """
-    # TODO(oschaaf): When we are on python 3 use RAII here.
-    sock = socket.socket(self.socket_type, socket.SOCK_STREAM)
-    sock.bind((address, 0))
-    port = sock.getsockname()[1]
-    sock.close()
+    with socket.socket(self.socket_type, socket.SOCK_STREAM) as sock:
+      sock.bind((address, 0))
+      port = sock.getsockname()[1]
+      sock.close()
```
Comment (htuch) [inline integration/integration_test_fixtures.py]: No need for `close`.

File: integration/nighthawk_test_server.py [hunk 3]
```diff
@@ -48,24 +48,24 @@ def serverThreadRunner(self):
...
-    return False
+  def waitUntilServerListening(self):
+    with socket.socket(self.socket_type, socket.SOCK_STREAM) as sock:
+      sock.settimeout(1)
+      tries = 10
+      while tries > 0:
+        time.sleep(0.5)
+        tries -= 1
+        if sock.connect_ex((self.server_ip, self.server_port)) == 0:
+          sock.close()
```
Comment (htuch) [inline integration/nighthawk_test_server.py]: No need for any `close`.

File: integration/BUILD (hunk 1 above)
Comment (oschaaf) [reply integration/BUILD]: Removed

**Pull request merged**